nca/Parsers/CalicoPolicyYamlParser.py

```diff
@@ -364,32 +392,61 @@ def _parse_icmp(self, icmp_data, not_icmp_data):
         allowed_keys = {'type': 0, 'code': 0}
         if icmp_data is not None:
             self.check_fields_validity(icmp_data, 'ICMP', allowed_keys)
-            err = ICMPDataSet.check_code_type_validity(icmp_type, icmp_code)
+            err = self.check_icmp_code_type_validity(icmp_type, icmp_code)
             if err:
                 self.syntax_error(err, icmp_data)
         if not_icmp_data is not None:
             self.check_fields_validity(not_icmp_data, 'notICMP', allowed_keys)
-            err = ICMPDataSet.check_code_type_validity(not_icmp_type, not_icmp_code)
+            err = self.check_icmp_code_type_validity(not_icmp_type, not_icmp_code)
             if err:
                 self.syntax_error(err, not_icmp_data)
 
-        res = ICMPDataSet(icmp_data is None and not_icmp_data is None)
+        protocols = ProtocolSet.get_protocol_set_with_single_protocol(protocol)
+        base_peer_set = self.peer_container.get_all_peers_group()
```

Comment:
what if `src_pods` or `dst_pods` contains ip-blocks, which are not part of `base_peer_set`?